### PR TITLE
move metadraw functions to new project

### PR DESCRIPTION
**Pull request by rmillikin**
#1980 broke Docker, because EngineLayer contained MetaDraw stuff which requires WPF which requires Windows. so I moved the MetaDraw stuff to a new project called GUIFunctions. It should build fine and be covered by unit tests but not crash Linux runs.

this PR also updates mzLib and related unit tests.

### CODE REVIEW

**Review by rmmiller22 — APPROVED** [review]

**Comment by trishorts** [comment]
can I safely assume that what I copied below is just moved code and not new? Therefore not needing additional testing?

image

File: Test/FdrTest.cs [hunk 1]
```diff
@@ -273,8 +273,8 @@ public static void TestComputePEPValue()
 
             var myMsDataFile = myFileManager.LoadFile(origDataFile, CommonParameters);
             var searchModes = new SinglePpmAroundZeroSearchMode(5);
-            List<Protein> proteinList = ProteinDbLoader.LoadProteinFasta(Path.Combine(TestContext.CurrentContext.TestDirectory, @"TestData\hela_snip_for_unitTest.fasta"), true, DecoyType.Reverse, false, ProteinDbLoader.UniprotAccessionRegex, ProteinDbLoader.UniprotFullNameRegex, ProteinDbLoader.UniprotFullNameRegex, ProteinDbLoader.UniprotGeneNameRegex,
-                    ProteinDbLoader.UniprotOrganismRegex, out var dbErrors, -1);
+            List<Protein> proteinList = ProteinDbLoader.LoadProteinFasta(Path.Combine(TestContext.CurrentContext.TestDirectory, @"TestData\hela_snip_for_unitTest.fasta"), true, DecoyType.Reverse, false, out var dbErrors, ProteinDbLoader.UniprotAccessionRegex, ProteinDbLoader.UniprotFullNameRegex, ProteinDbLoader.UniprotFullNameRegex, ProteinDbLoader.UniprotGeneNameRegex,
```
Comment (trishorts) [inline Test/FdrTest.cs:276]: Does out var dbErrors need unit testing? I didn't see one added in this PR.

Comment (rmillikin) [reply Test/FdrTest.cs:276]: IMO it doesn't really need one urgently, they're more like warnings. at some point we could add one though

**Review by trishorts — APPROVED** [review]
I built. tested the gui with the vignette. I also ran a search in linux with one .raw file and a db.gz.

Would like to see a unit test for out var dbErrors in a future PR.

**Pull request merged**